### PR TITLE
Fix some warnings 3.9

### PR DESCRIPTION
**Pull request by MBkkt**
### Scope & Purpose

Fix correct warnings

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*
  - [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

**Comment by KVS85** [comment]
- [ ] https://jenkins.arangodb.biz/job/arangodb-ANY-mac-matrix.x86-64-2/8/

**Comment by MBkkt** [comment]
https://jenkins.arangodb.biz/job/arangodb-ANY-mac-matrix.x86-64-2/9/

File: 3rdParty/iresearch/external/absl/absl/functional/function_ref.h [hunk 1]
```diff
@@ -121,14 +121,8 @@ class FunctionRef<R(Args...)> {
...
   // Typically, it should only be used as an argument type.
-  #if defined(__APPLE__) && __clang_major__ == 13 && __clang_minor__ == 1
-    #pragma clang diagnostic push
-    #pragma clang diagnostic ignored "-Wdeprecated-copy"
-  #endif
   FunctionRef& operator=(const FunctionRef& rhs) = delete;
-  #if defined(__APPLE__) && __clang_major__ == 13 && __clang_minor__ == 1
-    #pragma clang diagnostic pop
-  #endif
+  FunctionRef(const FunctionRef& rhs) = default;
```
Comment (MBkkt) [inline 3rdParty/iresearch/external/absl/absl/functional/function_ref.h:125]: https://github.com/abseil/abseil-cpp/blob/master/absl/functional/function_ref.h#L128

**Comment by MBkkt** [comment]
https://jenkins.arangodb.biz/job/arangodb-ANY-mac-matrix.x86-64-2/10/

**Comment by KVS85** [comment]
- [ ] https://jenkins.arangodb.biz/job/arangodb-ANY-mac-matrix.x86-64-2/12/

**Review by KVS85 — APPROVED** [review]
LGTM.

**Review by alexbakharew — APPROVED** [review]
LGTM

**Pull request merged**